### DISCUSSION TITLE
Vehicle Showing Incorrect SoC?

### DISCUSSION THREAD

**Discussion by ryan-hummer** [teslamate-org/teslamate#2478] [General]
Hi there,

This isn't quite TeslaMate specific, but I noticed this using TeslaMate. So Curious if anyone has seen similar behavior.

I started noticing with our 2021 MY after I charged the vehicle overnight, I open the app to turn on climate before leaving (not using schedule departure as WFH currently so random leaving times). I'd notice on the app the car would say a charging session had started after it reached 70% (the limit I set). The vehicle said it was down to 65% and wanted to charge.  I'd then go for a drive and sometime after that drive, the car would say the SoC was over 70%, like it lost track of that ~5%, then magically remembered it was there.

Using TeslaMate I was digging into it a bit more and did notice some of these repeat charge sessions and loss and regain of SoC values..

I have a service appointment scheduled, and server is wanting to question the 3rd party app installed. So far I've disabled it at their request and will charge the car overnight again to see if the behavior exhibits again. But we have two Tesla's and only this one shows this behavior.  I've been running TeslaMate for 2+ years already.

Here's an example image of what I'm seeing. Two stories in one...
![image](https://user-images.githubusercontent.com/88110754/154575083-8d52af5f-0180-443c-9867-19afdcf54781.png)

The charging cycles from 02:30 to just after 06:00, are what I've been seeing.  The 03:30 to just after 04:00 is the scheduled charge, then less than an hour later the car says it lost 5% range and charges up again to 70%, it does it two more times, until I leave.  I do some opportunistic charging from 06:30 to close to 10:30. I set the charge limit to 90%, but if you notice it starts to taper. Like the car is over 90% and charging to 100%.  I moved the car at the cut off time and it reported the >80% range.

Next is an image after I moved the car after the charge session..

![image](https://user-images.githubusercontent.com/88110754/154576126-b0f55d69-3b25-4135-b36c-c70b637c2cf5.png)

As you can see from the second image, I moved the car it then jumped to say it was over 90% charged and eventually said it was 100% charged.  Very interesting issue IMO.  Service so far says they don't see anything in their logs.

Again, just curious if anyone has seen something like this.
Thanks!

**Comment by init6-git** [comment]
What was the temperature? Is this to do with the car holding back some capacity when it's cold?

**Comment by ryan-hummer** [comment]
It is cold out, but the car does not show the snow flake icon. Looking at the data in Grafana the delta between the usable and available percentage is the same when I notice this "sudden change."
